Wrappers/Python/cil/optimisation/operators/Operator.py

```diff
@@ -188,15 +216,21 @@ def PowerMethod(operator, max_iteration=10, initial=None, tolerance = 1e-5,  ret
         2.0005647295658866
 
         """
+        convergence_check = True
+        
+        if method not in ["auto","direct_only","composed_with_adjoint"]:
+            ValueError(" The flag 'method' should take one of 'auto','direct_only','composed_with_adjoint'")
 
-        # Default case: non-symmetric
-        symmetric = False
-        try:
-            if operator.domain_geometry()==operator.range_geometry():
-                symmetric = True
-        except AssertionError:
-            # catch AssertionError for SIRF objects https://github.com/SyneRBI/SIRF-SuperBuild/runs/5110228626?check_suite_focus=true#step:8:972
-            pass
+        if method=="auto":
+            try:
+                if operator.domain_geometry() == operator.range_geometry():
+                    method="direct_only"
+                else:
+                    method="composed_with_adjoint"
+            except AssertionError:
+                # catch AssertionError for SIRF objects https://github.com/SyneRBI/SIRF-SuperBuild/runs/5110228626?check_suite_focus=true#step:8:972
+                method="composed_with_adjoint"
+        

```

Comment:
As we are enforcing only 2 choices from now, and the string is checked inside the loop I think we need to reassign them here to boolean which is a faster check. I've introduced the boolean `adjoint`. Feel free to rename/restructure, this is just a suggestion. The code may also not run so check it (especially the white space!)

Could be simplified to something like this:

```suggestion
            adjoint = True
            
            if method == "direct_only":
                adjoint = False
            elif method=="auto":
                try:
                    geometries_match = operator.domain_geometry() == operator.range_geometry

                except AssertionError:
                # catch AssertionError for SIRF objects https://github.com/SyneRBI/SIRF-SuperBuild/runs/5110228626?check_suite_focus=true#step:8:972
                     pass
                else:
                    if geometries_match:
                        adjoint=False

```